src/tbb/thread_data.h

```diff
@@ -40,6 +40,69 @@ class arena_slot;
 class task_group_context;
 class task_dispatcher;
 
+struct context_list {
+    std::size_t m_references{1};
+
+    //! Head of the thread specific list of task group contexts.
+    d1::context_list_node head{};
+
+    //! Last state propagation epoch known to this thread
+    /** Together with the_context_state_propagation_epoch constitute synchronization protocol
+    that keeps hot path of task group context construction destruction mostly
+    lock-free.
+    When local epoch equals the global one, the state of task group contexts
+    registered with this thread is consistent with that of the task group trees
+    they belong to. **/
+    std::atomic<std::uintptr_t> epoch{};
```

Comment:
Are `m_*` privates fields? Does it make sense to move them under `private` section?

Reply:
I think the only field that can be moved under `private` section is `m_references`. Other fields are used directly in several places in `task_group_context.cpp` (lock and traverse, updating of `epoch`). Can't tell if `context_list` deserves some complication.

Reply:
What about `m_mutex`? My idea is to combine `m_mutex` and `m_references` into private section while `head` and `epoch` remain untouched.

Reply:
`m_mutex` is also used directly in `thread_data::propagate_task_group_state` to traverse list.